src/content_cao.cpp

```diff
@@ -1814,6 +1832,30 @@ bool GenericCAO::directReportPunch(v3f dir, const ItemStack *punchitem,
 	return false;
 }
 
+bool GenericCAO::isNotItemStack() const
+{
+	// We know what is needed for something *not* to be
+	// an itemstack. However, we don't know for sure
+	// whether anything *is* an itemstack.
+	// (hence the check for 'not an itemstack')
+	static const v3f zero(0,0,0);
+	return
+		false
+		// Item stack may be physical or non-physical...
+		//|| m_prop.physical
+		|| m_prop.collideWithObjects
+		|| !m_is_visible
+		|| m_prop.makes_footstep_sound
+		|| m_prop.automatic_rotate == 0
+		|| m_prop.stepheight != 0
+		|| m_prop.automatic_face_movement_dir
+		// Item stack may be falling down first
+		//|| m_velocity != zero
+		// Acceleration may be zero or non-zero
+		//|| m_acceleration != zero
+		|| m_prop.visual != "wielditem";
```

Comment:
There is an `m_name` member, but it is empty for itemstacks. It is used for the player name, so it would be an instance name, instead of a class name.